astropy/cosmology/parameter/_core.py

```diff
@@ -275,8 +308,10 @@ def clone(self, **kw):
     def __repr__(self) -> str:
         """Return repr(self)."""
         fields_repr = (
+            # Get the repr, using the input fvalidate over the processed value
             f"{f.name}={(getattr(self, f.name if f.name != 'fvalidate' else '_fvalidate_in'))!r}"
             for f in fields(self)
-            if f.repr
+            # Only show fields that should be displayed and are not sentinel values
+            if (f.repr and (f.name != "default" or self.default is not MISSING))
```

Comment:
The outer parenthesis should be removed, but once that's done the pull request can be merged.
```suggestion
            if f.repr and (f.name != "default" or self.default is not MISSING)
```

Reply:
👍